pkg/clusterapi/name.go

```diff
@@ -25,6 +31,19 @@ func IncrementName(name string) (string, error) {
 	return ObjectName(match[1], n+1), nil
 }
 
+// IncrementNameWithFallbackDefault calls the IncrementName and fallbacks to use the default name if IncrementName
+// returns an error. This method is used to accommodate for any objects with name breaking changes from a previous version.
+// For example, in beta capi snowmachinetemplate is named after the eks-a snowmachineconfig name, without the '-1' suffix.
+// We set the object name to the default new machinetemplate name after detecting the invalid old name.
+func IncrementNameWithFallbackDefault(name, defaultName string) string {
+	n, err := IncrementName(name)
+	if err != nil {
+		logger.V(3).Info("Unable to increment object name (might due to changes of name format), fallback to the default name", "error", err.Error())
```

Comment:
nit nit: can we bump this to at least 4
That's our first "debug for devs" level